backend/webserver/views.py

```diff
@@ -239,6 +242,50 @@ def get(self, request):
         return Response(serializer.data, status=status.HTTP_200_OK)
 
 
+class LikedPostView(APIView):
+    authentication_classes = [BasicAuthentication]
+
+    def get_author(self,pk):
+        author = get_object_or_404(Author,pk=pk)
+        return author
+    
+    def get(self,requst,pk,*args,**kwargs):
+        author = self.get_author(pk)
+        public_post = Post.objects.filter(visibility="PUBLIC").all()
+        liked = Like.objects.filter(liker=author.id).filter(post__in=public_post.values_list('id')).all()
```

Comment:
```suggestion
        likes = Like.objects.filter(post__visibility="PUBLIC").filter(liker=author).all()
```
Can you try this?

Variable should also be named `likes` not `liked`.